bus/explorer.go

```diff
@@ -0,0 +1,81 @@
+package bus
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"go.sia.tech/renterd/api"
+)
+
+type (
+	// An explorer retrieves data about the Sia network from an external source.
+	Explorer interface {
+		Enabled() bool
+		BaseURL() string
+		SiacoinExchangeRate(ctx context.Context, currency string) (rate float64, err error)
+	}
+
+	explorer struct {
+		enabled bool
+		url     string
+	}
+)
+
+// NewExplorer returns a new Explorer.
+func NewExplorer(url string, enabled bool) Explorer {
+	return &explorer{
+		enabled: enabled,
+		url:     url,
+	}
+}
+
+// BaseURL returns the base URL of the Explorer.
+func (e *explorer) BaseURL() string {
+	return e.url
+}
+
+// Enabled returns true if the explorer is enabled.
+func (e *explorer) Enabled() bool {
+	return e.enabled
+}
+
+// SiacoinExchangeRate returns the exchange rate for the given currency.
+func (e *explorer) SiacoinExchangeRate(ctx context.Context, currency string) (rate float64, err error) {
+	// return early if the explorer is disabled
+	if !e.enabled {
+		return 0, api.ErrExplorerDisabled
+	}
+
+	// create request
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("%s/exchange-rate/siacoin/%s", e.url, currency), http.NoBody)
+	if err != nil {
+		return 0, fmt.Errorf("failed to create request: %w", err)
+	}
+	req.Header.Set("Accept", "application/json")
+
+	// create http client
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
Might be worthwhile to extract that into utils since we probably do that more than once. Including the status code check.

Reply:
OK, I'll F/U with this in a separate PR. Quite some occurrences indeed.